core/jazz_asset-event-handler/test/test.js

```diff
@@ -22,48 +22,544 @@ const chaiAsPromised = require('chai-as-promised'); chai.use(chaiAsPromised);
-    var getTokenRequest = index.getTokenRequest(configData);
+  describe("getTokenRequest", function () {
+    it("should return Request token when called", () => {
+      let result = index.getTokenRequest(configData);
+      expect(result.uri).to.eq(configData.BASE_API_URL + configData.TOKEN_URL);
+      expect(result.method).to.eq('post');
+    });
+  })
+
+  describe("getAuthResponse", function () {
+    it("should return auth response when called", () => {
+      let test = {
+        statusCode:200,
+        body:{
+          data:{
+            token:'sampletoken'
+          }
+        }
+      };
+      let result = index.getAuthResponse(test).then(result => {
+         expect(result).to.eq(test.body.data.token);
+      });
+    });
+
+    it("should give error message when fails", () => {
+      let test = {
+        statusCode:00,
+        body:{
+          data:{
+            token:'sampletoken'
+          }
+        }
+      };
+      let result = index.getAuthResponse(test).catch(error => {
+        expect(error.message).to.eq('Invalid token response from API');
+      });
+    });
+  })
+
+  describe('check if assets exists',() => {
+    it('assets found',() => {
+      let responseObject = {
+        statusCode: 200,
+        body: {
+          data: ["test1","test2"]
+          }
+      };
+      reqStub = sinon.stub(request, "Request").callsFake((obj) => {
+        return obj.callback(null, responseObject, responseObject.body);
+      });
+
+      let result = index.checkIfAssetExists(eventPayload, configData, authToken).then(obj =>{
+        sinon.assert.calledOnce(reqStub);
+        expect(obj).to.eq(responseObject.body.data[0]);
+        reqStub.restore();
+      });
+    });
+
+    it('assets not found',() => {
+      let responseObject = {
+        statusCode: 0,
+        body: {
+          data: [],
+          message:"No Assets found"
+        }
+      };
+      reqStub = sinon.stub(request, "Request").callsFake((obj) => {
+        return obj.callback(null, responseObject, responseObject.body);
+      });
+      let result = index.checkIfAssetExists(eventPayload, configData, authToken).catch(obj =>{
+        sinon.assert.calledOnce(reqStub);
+        expect(obj.details).to.eq('No Assets found');
+        reqStub.restore();
+      });
+    });
+
+    it('assets fails',() => {
+      let responseObject = null;
+      let errorObject = {
+        message:"",
+        type:"200"
+      };
+      reqStub = sinon.stub(request, "Request").callsFake((obj) => {
+        return obj.callback(errorObject, null, null);
+      });
+      let result = index.checkIfAssetExists(eventPayload, configData, authToken).catch(obj =>{
+        sinon.assert.calledOnce(reqStub);
+        expect(obj.type).to.eq('200');
+        reqStub.restore();
+      })
+    });
+  })
+
+  describe("processUpdateAsset", function () {
+    let responseObject,record;
+		afterEach(() => {
```

Comment:
here as well. i think i mentioned that this to be fixed in all the places and not only where the comment is put

Reply:
updated